src/esp/assets/ResourceManager.cpp

```diff
@@ -83,28 +83,45 @@ Magnum::GL::AbstractShaderProgram* ResourceManager::getShaderProgram(
 
       case COLORED_SHADER: {
         shaderPrograms_[COLORED_SHADER] =
-            std::make_shared<Magnum::Shaders::Flat3D>(
-                Magnum::Shaders::Flat3D::Flag::ObjectId);
+            std::make_shared<Magnum::Shaders::Phong>(
+                Magnum::Shaders::Phong::Flag::ObjectId, 3 /*lights*/);
       } break;
 
       case VERTEX_COLORED_SHADER: {
         shaderPrograms_[VERTEX_COLORED_SHADER] =
-            std::make_shared<Magnum::Shaders::Flat3D>(
-                Magnum::Shaders::Flat3D::Flag::ObjectId |
-                Magnum::Shaders::Flat3D::Flag::VertexColor);
+            std::make_shared<Magnum::Shaders::Phong>(
+                Magnum::Shaders::Phong::Flag::ObjectId |
+                    Magnum::Shaders::Phong::Flag::VertexColor,
+                3 /*lights*/);
       } break;
 
       case TEXTURED_SHADER: {
         shaderPrograms_[TEXTURED_SHADER] =
-            std::make_shared<Magnum::Shaders::Flat3D>(
-                Magnum::Shaders::Flat3D::Flag::ObjectId |
-                Magnum::Shaders::Flat3D::Flag::Textured);
+            std::make_shared<Magnum::Shaders::Phong>(
+                Magnum::Shaders::Phong::Flag::ObjectId |
+                    Magnum::Shaders::Phong::Flag::DiffuseTexture,
+                3 /*lights*/);
       } break;
 
       default:
         return nullptr;
         break;
     }
+
+    /* Default setup for Phong, shared by all models */
+    if (type == COLORED_SHADER || type == VERTEX_COLORED_SHADER ||
+        type == TEXTURED_SHADER) {
+      using namespace Magnum::Math::Literals;
+
+      static_cast<Magnum::Shaders::Phong&>(*shaderPrograms_[TEXTURED_SHADER])
+          .setLightPositions({Magnum::Vector3{10.0f, 10.0f, 10.0f} * 100.0f,
```

Comment:
Are these light positions randomly picked?
My idea is to put the lights on the corners of the bounding box of the scene, which can be adaptive.

Reply:
Yes, those are just a verbatim copy from [magnum-player](https://magnum.graphics/showcase/player). There it was hand-tuned to give a nicely balanced lighting (and I think it's not bad here either), but something else could fit here even better.

The lights are currently relative to the camera so you always get the scene uniformly lit -- not sure if making them fully static would achieve the same, maybe balance them out? Definitely might need some fine-tuning :)

Reply:
> The lights are currently relative to the camera so you always get the scene uniformly lit -- not sure if making them fully static would achieve the same, maybe balance them out? Definitely might need some fine-tuning :)

I may misunderstand you:
When the camera moves, the lights follow?

I think it might be more realistic if the lights are fixed (though the scene looks ugly from some angles), especially if we would like to add shadows to the objects in the scene (we talked about it recently).

Reply:
> When the camera moves, the lights follow?

Yes, because that looked like the best tradeoff for me -- as there are no shadows, having a room affected by a light coming through walls from some other room will not look realistic *at all*. 

IIRC back in April we talked about some models having designated light positions that could be used for giving the lights static positions? That could make sense to do together with shadows.

This PR is quite experimental, adding a completely new feature where it's unclear how it should be used. So please clone it, play with it, change light positions / colors / ... to see what fits best. I won't like to merge it without having a consensus on how the visuals should look like :)